MapboxGeocoderTests/BatchGeocodingTests.swift

```diff
@@ -0,0 +1,384 @@
+            CLLocationCoordinate2DMake(100, 100),
+            CLLocationCoordinate2DMake(100, 100)
+        ]
+        
+        let options = ReverseBatchGeocodeOptions(coordinates: queries)
+        
+        let task = geocoder.batchGeocode(options) { (placemarks, attribution, error) in
+            
+            let queries = placemarks!
+            let attribution = attribution![0]
+            
+            XCTAssertEqual(queries.count, 3, "reverse batch geocode should have 3 queries")
+            
+            for result in queries {
+                XCTAssertEqual(result.count, 0, "each reverse batch geocode query should have 0 found results")
+                XCTAssertEqual(attribution, "© 2017 Mapbox and its suppliers. All rights reserved. Use of this data is subject to the Mapbox Terms of Service. (https://www.mapbox.com/about/maps/)")
+            }
+            
+            expectation.fulfill()
+        }
+        XCTAssertNotNil(task)
+        
+        waitForExpectations(timeout: 1) { (error) in
+            XCTAssertNil(error, "Error: \(error!)")
+            XCTAssertEqual(task.state, URLSessionTask.State.completed)
+        }
+    }
+    
+    /**
+     General batch geocoding tests - invalid queries, invalid tokens, token scope checking, etc.
+     */
+    
+    func testInvalidBatchGeocode() {
+        _ = stub(condition: isHost("api.mapbox.com")
+            && isPath("/geocoding/v5/mapbox.places-permanent////.json")
+            && containsQueryParams(["access_token": BogusToken])) { _ in
+                let path = Bundle(for: type(of: self)).path(forResource: "permanent_invalid", ofType: "json")
+                return OHHTTPStubsResponse(fileAtPath: path!, statusCode: 200, headers: ["Content-Type": "application/vnd.geo+json"])
+        }
+        
+        let expectation = self.expectation(description: "invalid batch geocoding query should not return results")
+        let geocoder = Geocoder(accessToken: BogusToken)
+        let options = ForwardBatchGeocodeOptions(query: "///")
+        let task = geocoder.batchGeocode(options) { (placemarks, attribution, error) in
+            
+            XCTAssertEqual(error!.localizedFailureReason, "Not Found")
+            
+            expectation.fulfill()
+        }
+        XCTAssertNotNil(task)
+        
+        waitForExpectations(timeout: 1) { (error) in
+            XCTAssertNil(error, "Error: \(error!)")
+            XCTAssertEqual(task.state, URLSessionTask.State.completed)
+        }
+    }
+    
+    func testInvalidTokenForBatchGeocode() {
+        let invalidToken = "xyz"
+        
+        _ = stub(condition: isHost("api.mapbox.com")
+            && isPath("/geocoding/v5/mapbox.places-permanent/85+2nd+st+san+francisco.json")
+            && containsQueryParams(["access_token": invalidToken])) { _ in
+                let path = Bundle(for: type(of: self)).path(forResource: "permanent_invalid_token", ofType: "json")
```

Comment:
Tests are failing:

```
Testing failed:
	error: /Users/Desktop/permanent_invalid_token.json: No such file or directory
	error: /Users/Desktop/permanent_reverse_multiple_no_results.json: No such file or directory
```

Did you add these files to the iOS and tvOS test targets as well? (There’s no watchOS test target.)

Reply:
Looks like I forgot to `Copy items if needed` when adding these fixtures to the project 😬